### DISCUSSION TITLE
YOLOv8 workflow

### DISCUSSION THREAD

**Discussion by Shiwo0826** [ultralytics/ultralytics#10215] [Q&A]
When using YOLOv8 for segmentation tasks, should semantic segmentation be implemented first or target detection implemented first, or both at the same time?

**Comment by pderrenger** [comment]
@MuHe0826 in YOLOv8 workflow for segmentation tasks, both semantic segmentation and target detection can effectively be implemented at the same time thanks to YOLOv8's integrated approach. It allows you to run segmentation and detection concurrently, taking advantage of YOLOv8's architecture which is designed to handle multiple computer vision tasks in a unified manner. This means you're not required to prioritize one over the other; instead, you can leverage YOLOv8's capabilities to perform both tasks simultaneously for efficient and robust results. Here's an example of how you might initiate your model for such tasks:

```python
from ultralytics import YOLO

# Load your YOLOv8 model
model = YOLO('yolov8-segment.pt')

# Perform detection and segmentation
results = model('path/to/image.jpg')
```

This snippet assumes you have a YOLOv8 model trained for segmentation (`yolov8-segment.pt`). The model will output both detection and segmentation results for the input image. Happy coding! 😊

**Comment by Shiwo0826** [comment]
@pderrenger Thank you very much for your answer!

**Reply by pderrenger** [reply]
You're welcome! If you have any more questions or need further assistance, feel free to ask. Happy coding! 😊
